src/usePersistedState.js

```diff
@@ -30,19 +25,23 @@ const usePersistedState = (initialState, key, { get, set }) => {
     return () => {
       globalState.current.deregister();
     };
-  }, []);
+  }, [initialState, key]);
```

Comment:
Because of this change, I will probably update this (only place used) so the empty array is wrapped in `useMemo`, but I think the rest should be the same:

https://github.com/Bridebook/Bridebook-next/blob/master/packages/toolbox/src/storage-managers/managers/array-storage-hook-manager.ts#L21